File: QuantConnect.AlphaStream/Requests/SubscribeRequest.cs
using QuantConnect.AlphaStream.Infrastructure;
using QuantConnect.AlphaStream.Models;
using RestSharp;

namespace QuantConnect.AlphaStream.Requests
{
    /// <summary>
    /// Subscribe to an Alpha given by id for a month and charge the subscription the funds account.
    /// </summary>
    [Endpoint(Method.POST, "alpha/{id}/subscribe")]
    public class SubscribeRequest : AttributeRequest<ApiResponse>
    {
        /// <summary>
        /// Identifier of the Alpha to subscribe.
        /// </summary>
        [PathParameter("id")]
        public string Id { get; set; }

        /// <summary>
        /// Try and subscribe to the Alpha with an exclusive preference.
        /// </summary>
        [QueryParameter("exclusive")]
        public bool Exclusive { get; set; }
    }
}

File: QuantConnect.AlphaStream.Python/Requests/SubscribeRequest.py


class SubscribeRequest(object):
    """ Send subscription request for an Alpha """

    def __init__(self, alphaId, exclusive=False):
        self.Id    = str(alphaId)
        self.Endpoint   = "alpha/" + self.Id + "/subscribe"
        self.Exclusive  = exclusive
        
    def GetPayload(self):
        payload = {
            "id"        : self.Id,
            "exclusive" : self.Exclusive
        }
        return payload